graphdatascience/graph/nx_loader.py

```diff
@@ -0,0 +1,124 @@
+from collections import defaultdict
+from typing import Any, Dict, List, Set, Tuple
+
+import networkx as nx
+from pandas import DataFrame
+
+from ..error.client_only_endpoint import client_only_endpoint
+from ..error.illegal_attr_checker import IllegalAttrChecker
+from ..error.uncallable_namespace import UncallableNamespace
+from ..server_version.compatible_with import compatible_with
+from ..server_version.server_version import ServerVersion
+from .graph_object import Graph
+
+
+class NXLoader(UncallableNamespace, IllegalAttrChecker):
+    @client_only_endpoint("gds.graph.networkx")
+    @compatible_with("load", min_inclusive=ServerVersion(2, 1, 0))
+    def load(self, nx_G: nx.Graph, graph_name: str, concurrency: int = 4) -> Graph:
+        nodes, rels = self._parse(nx_G)
+
+        undirected_rel_types = []
+        if not isinstance(nx_G, nx.DiGraph) and not isinstance(nx_G, nx.MultiDiGraph):
```

Comment:
```suggestion
        if nx_G.to_directed_class():
```
reading at https://networkx.org/documentation/stable/reference/classes/multigraph.html this check would allow us to handle also custom implementations of the NetworkX API. (should be None if its directed)

Reply:
Cool! Will check this out

Reply:
If it's directed it will return itself, it won't be None. But yes, probably the safest check would be:
```
if not isinstance(nx_G.to_directed_class(), type(nx_G)):
```
If they haven't overloaded the method `to_directed_class` in their custom graph, it will default to returning something like `nx.DiGraph`, but that's their problem and really nothing we can do I think

Reply:
done